feat(custom-views): Add hook for fetching issue counts

MichaelSun48 opened the pull request:
Adds a useAPIQuery hook for the `organization/.../issues-count` endpoint. This is going to be used by the custom views component to fetch the query counts for new tabs. 

Update 1/6: We've decided to re-add this functionality, which is why this PR has been reopened

malwilley commented on static/app/views/issueList/queries/useFetchIssueCounts.tsx:
```diff
@@ -0,0 +1,30 @@
+  options: Partial<UseApiQueryOptions<GroupSearchView[]>> = {}
+) => {
+  return useApiQuery<GroupSearchView[]>(makeFetchIssueCounts(params), {
+    staleTime: Infinity,
```
Counts seem likely to change, so I'd make this `staleTime: 0` or something small like 10 seconds

malwilley commented on static/app/views/issueList/overview.tsx:
```diff
@@ -139,7 +139,7 @@ interface EndpointParams extends Partial<PageFilters['datetime']> {
 }
 
-type CountsEndpointParams = Omit<EndpointParams, 'cursor' | 'page' | 'query'> & {
+export type CountsEndpointParams = Omit<EndpointParams, 'cursor' | 'page' | 'query'> & {
```
I'd either duplicate this type in `useFetchIssueCounts` or add it there and export here. Generally we don't want utils importing from components (should be the other way around)

gggritso committed:
feat(dashboards): Improve `WidgetFrame` error handling (#82759)

- Don't allow full screen in error state
- Show actions in error mode if there is no retry
- Add error boundary

Zylphrex committed:
chore(explore): Default to RPC behind feature flag (#82589)

This switches explore to use the RPC layer to make queries by default
behind a feature flag.

ryan953 committed:
fix(hydration issue): Disable focusTrap when using a hovercard inside hydration issue modal (#82779)

These links will accept clicks now:
<img width="559" alt="SCR-20241231-kcqh"
src="https://github.com/user-attachments/assets/4f3e4ec3-0033-4ced-8c53-dbfdcf445af6"
/>


Fixes #82776

Zylphrex committed:
chore(typing): Add typing to datasets/metrics.py (#82781)

edwardgou-sentry committed:
feat(insights): adds flag for handling missing vitals (#82762)

Currently, if a specific web vital is never reported in a query, we
treat the score as 0. This is actually misleading, instead we want to
exclude missing vitals from any score calculation. This flag will
control the logic in the frontend and backend to handle that.

asottile-sentry committed:
ref: fix missing coverage due to typo in test (#82788)

<!-- Describe your PR here. -->

asottile-sentry committed:
ref: first pass at removing some dead code in tests (#82792)

I only got through about ~30% of the test files -- will follow up with
other ones

some explanation for some patterns:
- deleted function => nothing called it
- `raise SomeType` -> `raise AssertionError` -- `AssertionError` is
excluded from coverage
- `pass` -> `raise NotImplementedError` -- these functions are never
called -- only used for their shape / name and `NotImplementedError` is
excluded from coverage
- `try: some_test_code except: fail(...)` -> no need for the `try`, an
uncaught exception already fails the test

<!-- Describe your PR here. -->

ryan953 committed:
test: Fix testing-library/prefer-screen-queries violations (#82786)

https://github.com/testing-library/eslint-plugin-testing-library/blob/main/docs/rules/prefer-screen-queries.md

ameliahsu committed:
chore(onboarding): remove members invite members ff from backend (#82796)

remove `members-invite-teammates` feature flag from backend

ameliahsu committed:
chore(onboarding): remove members invite members ff from frontend (#82797)

remove `members-invite-teammates` feature flag from frontend

Commit:
feat(devservices): Docker compose backend tests use devservices (#81941)

This allows us to measure the stability of devservices for sentry
backend tests. Changing the cadence of this scheduled job to hourly
again, as we need more data here. Will check devinfra metrics to get a
better sense of how well this is working in CI

Commit:
chore(devservices): Bump devservices to 1.0.8 (#82799)

picks up https://github.com/getsentry/devservices/releases/tag/1.0.8

JonasBa committed:
ref(tsc) fix nouncheckedindexedaccess (#82791)

Precursor to enabling ts noUncheckedIndexedAccess rule. The fixes here
dont impact the runtime and only add type assertions. I will followup
with a PR to do the same for getsentry before enabling the flag

vgrozdanic committed:
fix(onboarding): Support for multiword frameworks (#82805)

So far we had a bug when the framework would contain multiple words,
e.g. `java-spring-boot` would only return `spring` instead of
`spring-boot` as a framework name.

---------

Signed-off-by: Vjeran Grozdanic <vjeran.grozdanic@sentry.io>

iamrajjoshi committed:
:wrench: chore(api): move OpenApiParameters definition to parameters.py (#82801)

asottile-sentry committed:
ref: is_metrics_based is always True (#82787)

<!-- Describe your PR here. -->

ryan953 committed:
ref(eslint): Consolidate react related rules (#82783)

There's two commits in here: 
**1st commit**: just copy+paste all the rules we had explicitly defined
into one spot. I moved the plugin definitions and the recommended set
into there too.
**2nd commit** delete any rule definitions that are either included in
the recommended config and are 'error' by default. Also deleted some
that are not in the recommended set and are 'off' by default. Overall
this make the config more readable, the more we can use recommended
configs the simpler the config, and more 'typical' our app will be.


If you want to see a nice list of the rules, run `npx eslint
--inspect-config .` then find the section called `react` and open it up.
You can see what's on/off overall, either from the recommended set, or
overridden after. It's something like this:
<img width="1294" alt="SCR-20241231-kphm"
src="https://github.com/user-attachments/assets/9258b782-f719-461b-9d5a-7011ab58d2c2"
/>

michellewzhang committed:
ref(replay): make configure replay card mobile-specific (#82120)

closes https://github.com/getsentry/sentry/issues/72231

<img width="1175" alt="SCR-20241213-psnz"
src="https://github.com/user-attachments/assets/ba0046cd-ae10-472d-af30-25fd99067f7d"
/>


the web replay configure links go to the `javascript` docs links, so
they're not platform-specific. we are assuming the user will change
their platform in the docs as needed. i chose to use the `react-native`
docs links for the mobile card since it's difficult to get the exact
platform information from the replay.

michellewzhang committed:
feat(flags): add unleash webhook (#82500)

relates to https://github.com/getsentry/team-replay/issues/515

todo

- [x] add validation with auth header
- [x] add tests
- [x] get ngrok test working (currently getting `404 not found`)

michellewzhang committed:
chore(replay): add mobile beta org flag to sentry (#82654)

Enable GA banner for mobile replay beta orgs about grace period that
will last 2 months. Flag can be removed after March 7th 2024.

**there will be a separate flag for bypassing billing. cc @cmanallen**

relates to 
- https://github.com/getsentry/team-replay/issues/523

roggenkemper committed:
chore(inbound-filters): Update legacy browser filters  (#82060)

the goal of the new filters was to reevaluate them periodically to keep
them relevant, which is what the pr does. the actual changes happen in
https://github.com/getsentry/relay/pull/4381, but this will update the
inbound filters page to reflect what versions are being filtered.

ryan953 committed:
test: Fix testing-library/no-container violations (#82811)

Fixed some, ignored some others. It's more important to get this rule
enabled

scttcper committed:
feat(settings): Add project avatar to team dropdown (#82790)

roggenkemper committed:
chore(issue-details): Add event action analytics (#82812)

this pr improves or adds analytics for the open json, copy event id, and
copy event link actions on the issue details page

lynnagara committed:
perf: don't log exception for every stale message (#82553)

depends on https://github.com/getsentry/arroyo/pull/416

scttcper committed:
feat(ui): Remove findDomNode from SessionStack (#82638)

JonasBa committed:
fix(profiling) ellipsis on function names (#82806)

Fix overflow ellipsis on the flamegraph function names and prevent
wrapping

ameliahsu committed:
chore(onboarding): remove unused feature flags (#82816)

removing feature flags for
- members invite members
- messaging integration onboarding on the project creation page

malwilley committed:
chore(issue-stream): Add a feature flag for the issue stream component refactor (#82810)

scttcper committed:
feat(react19): Remove findDomNode in textCopyInput (#82637)

scttcper committed:
ref(performance): Remove unused single/double axis chart (#82674)

asottile-sentry committed:
ref: remove unused Plugin configure (#82813)

fixes a handful of errors in typing-ignored files

<!-- Describe your PR here. -->

ryan953 committed:
ref(eslint): Consolidate `import` rules (#82798)

These are all the recommended `import` rules:
https://github.com/import-js/eslint-plugin-import/blob/main/config/flat/recommended.js

But typescript has us covered for 5/6 of them:
https://typescript-eslint.io/troubleshooting/typed-linting/performance/#slow-eslint-rules

So we don't need the recommended set. Instead, lets just enable the ones
we want. And anything that's set to `'off'` we can remove from the
config.

evanpurkhiser committed:
ref(browserHistory): Remove from queuesTable (#82735)

evanpurkhiser committed:
ref(browserHistory): Remove from newAuthToken (#82734)

evanpurkhiser committed:
ref(browserHistory): Remove from domainsTable (#82733)

evanpurkhiser committed:
ref(browserHistory): Remove from screenLoadSpansTable (#82732)

evanpurkhiser committed:
ref(browserHistory): Remove from httpLandingPage (#82731)

evanpurkhiser committed:
ref(browserHistory): Remove from resourceView (#82730)

evanpurkhiser committed:
ref(browserHistory): Remove from pagePerformanceTable (#82729)

evanpurkhiser committed:
ref(browserHistory): Remove from releaseSelector (#82727)

evanpurkhiser committed:
ref(browserHistory): Remove from appStarts/spanOperationTable (#82726)

evanpurkhiser committed:
ref(browserHistory): Remove from eventSamplesTable (#82725)

evanpurkhiser committed:
ref(browserHistory): Remove from queuesLandingPage (#82724)

evanpurkhiser committed:
ref(browserHistory): Remove from webVitals/pageOverview (#82722)

evanpurkhiser committed:
ref(browserHistory): Remove from app/root (#82721)

evanpurkhiser committed:
ref(browserHistory): Remove from resourceSummaryTable (#82720)

MichaelSun48 committed:
chore(api): Delete GroupParticipants endpoint (#82815)

Deletes the `/<issue_id>/participants/` endpoint and all usages. 

I could not find any usages on the frontend of this endpoint, nor in any
of our logs. Please feel free to double check me on that

scttcper committed:
fix(eslint): Disable typescript server in eslint (#82817)

evanpurkhiser committed:
ref(browserHistory): Remove from renderBlockingSelector (#82719)

evanpurkhiser committed:
ref(browserHistory): Remove from resourceTable (#82718)

edwardgou-sentry committed:
feat(insights): Update backend performance score function to handle missing vitals (#82750)

Currently, if a webvital is missing, we treat the individual score as 0.
This can be misleading because it negatively affects the overall
performance score of an organization/project.

Instead, we want to remove any missing vital factors from the overall
performance score calculation, which is what this change aims to do by
updating the `performance_score` function.

scttcper committed:
feat(projects): Allow viewing all projects (#82115)

edwardgou-sentry committed:
feat(insights): score ring and breakdown chart drop missing vital segments (#82686)

Currently in the Web Vitals module, if a specific vital is never
reported in the organization or project, we just assume that the score
of the vital is 0. However, this is misleading to the user, since the
score is not actually 0.

Instead, we want to handle missing vitals by excluding them from and
score calculation. This means dropping missing vitals from the score
ring and rebalancing the score breakdown timeseries in the frontend.

michellewzhang committed:
fix(flags): fix secret post link (#82819)

it should be the org slug after `/organizations/`, we didn't catch that
the first time around

scttcper committed:
ref(ui):  use toSpliced in replaceAtArray (#82681)

ameliahsu committed:
feat(workflow-engine): add `EventAttributeConditionHandler` (#82741)

add `EventAttributeConditionHandler`

leeandher committed:
feat(issue-details): Allow filters for first/last/recommended (#82476)

This PR does a few things, but mostly addresses the inconsistencies with
filtering in the streamlined UI.

- Filters will now apply to the queries for First/Last/Recommended to
ensure better results for the loaded event, matching the user
expectations. These filters are env, date range, and queries.
- The missing event state has been redesigned to be more helpful,
letting users easily click to recommended/clear filters and adds a
graphic.
- The first/last/recommended buttons will now be accessible even when an
event is not loaded

**Specific event missing**

<img width="1213" alt="image"
src="https://github.com/user-attachments/assets/8ad3aedc-a159-4c51-b733-a693301b4779"
/>

**Query doesn't find anything**

<img width="1182" alt="image"
src="https://github.com/user-attachments/assets/9847c5c3-b6ef-4314-8c57-b032bc1da17e"
/>


**todo**
- [x] Add or update tests

michellewzhang committed:
ref(flags): don't show flag legend if no lines to show (#82826)

if there are no intersection flags, we won't show any lines -- let's
hide the feature flag label from the legend

cathteng committed:
nit(aci): don't json.dump or load for `comparison` JSONField (#82827)

evanpurkhiser committed:
ref(browserHistory): Remove from pipelinesTable (#82712)

evanpurkhiser committed:
ref(browserHistory): Remove from domainTransactionsTable (#82713)

evanpurkhiser committed:
ref(browserHistory): Remove from browserTypeSelector (#82715)

evanpurkhiser committed:
ref(browserHistory): Remove from transactionProfiles/content (#82716)

evanpurkhiser committed:
ref(browserHistory): Remove from queryTransactionsTable (#82717)

evanpurkhiser committed:
ref(browserHistory): Remove from cache/transactionsTable (#82728)

evanpurkhiser committed:
ref(browserHistory): Remove from releaseComparisonChart (#82629)

evanpurkhiser committed:
ref(browserHistory): Remove from transactionTags/content (#82628)

evanpurkhiser committed:
ref(browserHistory): Remove from useTraceQueryParamStateSync (#82627)

evanpurkhiser committed:
ref(browserHistory): Remove from organizationApiKeyDetails (#82626)

evanpurkhiser committed:
ref(browserHistory): Remove from transactionSpans/content (#82625)

evanpurkhiser committed:
ref(browserHistory): Remove from transactionVitals/content (#82624)

evanpurkhiser committed:
ref(browserHistory): Remove from transactionProfiles (#82623)

evanpurkhiser committed:
ref(browserHistory): Remove from transactionOverview (#82622)

evanpurkhiser committed:
ref(browserHistory): Remove from transactionEvents/content (#82620)

evanpurkhiser committed:
ref(browserHistory): Remove from transactionSelector (#82619)

evanpurkhiser committed:
ref(browserHistory): Remove from tagsDisplay (#82618)

Co-authored-by: Michelle Zhang <56095982+michellewzhang@users.noreply.github.com>
Co-authored-by: getsantry[bot] <66042841+getsantry[bot]@users.noreply.github.com>

evanpurkhiser committed:
ref(browserHistory): Remove from transactionOverview/charts (#82617)

evanpurkhiser committed:
ref(browserHistory): Remove from transactionOverview/content (#82616)

evanpurkhiser committed:
ref(browserHistory): Remove from SuspectSpansHeader (#82614)

evanpurkhiser committed:
ref(browserHistory): Remove form sentryApplicationDetails (#82613)

evanpurkhiser committed:
ref(browserHistory): Remove from spanMetricsTable (#82611)

evanpurkhiser committed:
ref(browserHistory): Remove from groupList (#82599)

evanpurkhiser committed:
ref(browserHistory): Remove from sortLink (#82600)

evanpurkhiser committed:
ref(browserHistory): Replace in durationChart (#82602)

ryan953 committed:
test: ignore testing-library/no-node-access violations (#82829)

evanpurkhiser committed:
ref(browserHistory): Remove from spanDetailsControls (#82605)

evanpurkhiser committed:
ref(browserHistory): Remove from exclusiveTimeTimeSeries (#82606)

ryan953 committed:
ci(eslint): Use the prettier-config to disable conflicting rules, prettier itself still runs and does the formatting (#82830)

evanpurkhiser committed:
ref(browserHistory): Remove from trendChart (#82607)

evanpurkhiser committed:
ref(browserHistory): Remove from statusBreakdown (#82608)

evanpurkhiser committed:
ref(browserHistory): Remove from spanDetails/chart (#82609)

evanpurkhiser committed:
ref(browserHistory): Remove from snoozeAlert (#82598)

evanpurkhiser committed:
ref(browserHistory): Remove from widgetContainer (#82711)

evanpurkhiser committed:
ref(browserHistory): Remove from platformSelector (#82710)

evanpurkhiser committed:
ref(browserHistory): Remove from startTypeSelector (#82709)

evanpurkhiser committed:
ref(browserHistory): Remove from deviceClassSelector (#82708)

evanpurkhiser committed:
ref(browserHistory): Remove from spanOpSelector (#82707)

evanpurkhiser committed:
ref(browserHistory): Remove from transactionsTable (#82706)

evanpurkhiser committed:
ref(browserHistory): Remove from projectCrumb (#82705)

evanpurkhiser committed:
ref(browserHistory): Remove from appStarts/spanOpSelector (#82704)

evanpurkhiser committed:
ref(browserHistory): Remove from screensView (#82703)

evanpurkhiser committed:
ref(browserHistory): Remove from authTokenDetails (#82702)

evanpurkhiser committed:
ref(browserHistory): Remove from groupEventDetails (#82701)

evanpurkhiser committed:
ref(browserHistory): Remove from screensLandingPage (#82700)

evanpurkhiser committed:
ref(browserHistory): Remove from screensTemplate (#82699)

evanpurkhiser committed:
ref(browserHistory): Remove from screenSummaryPage (#82698)

evanpurkhiser committed:
ref(browserHistory): Remove from queriesTable (#82697)

evanpurkhiser committed:
ref(browserHistory): Remove from slowestProfileFunctions (#82695)

evanpurkhiser committed:
ref(browserHistory): Remove from teamCrumb (#82694)

evanpurkhiser committed:
ref(browserHistory): Remove from screenDetailsPage (#82693)

evanpurkhiser committed:
ref(browserHistory): Remove from regressedProfileFunctions (#82634)

evanpurkhiser committed:
ref(browserHistory): Remove form actionSelector (#82633)

evanpurkhiser committed:
ref(browserHistory): Remove from domainSelector (#82631)

evanpurkhiser committed:
ref(browserHistory): Remove from transactionEvents (#82630)

evanpurkhiser committed:
fix(router): Add missing normalizeUrl to rr6 useNavigate (#82163)

scttcper committed:
fix(settings): Bookmark toggling wrong direction (#82832)

ryan953 committed:
ref(eslint): Consolidate `sentry` & `sort-order` rules (#82834)

evanpurkhiser committed:
ref(browserHistory): Remove from performance/searchBar (#82596)

evanpurkhiser committed:
ref(browserHistory): Remove from integrationListDirectory (#82615)

edwardgou-sentry committed:
fix(insights): Webvitals flag set api_expose to true (#82833)

evanpurkhiser committed:
feat(uptime): Adds a simple overview page to the uptime insights tab (#82567)

Looks like this

<img alt="clipboard.png" width="1710"
src="https://i.imgur.com/74rvSUI.jpeg" />

evanpurkhiser committed:
ref(browserHistory): Remove from teamSettings (#82736)

evanpurkhiser committed:
ref(browserHistory): Remove from organizationCrumb (#82714)

ryan953 committed:
ref(eslint): Consolidate typescript-eslint rules (#82839)

asottile-sentry committed:
ref: delete unused never_cache (#82837)

<!-- Describe your PR here. -->

evanpurkhiser committed:
fix(js): Correct this binding in groupList#handleCursorChange (#82840)

ryan953 committed:
ref(eslint): Consolidate eslint core rules (#82844)

leeandher committed:
ref(issue-details): Rework the shared issue details context (#82838)

Refactoring the shared issue details context. The only real change is
that the sidebar now always appears open when you land on the page
(rather than using local storage)
- Name change `useEventDetails` -> `useIssueDetails`
- Rather than a generic `updateState` action, now has a specific
`UpdateNavScrollMargin` action to avoid messing with other state keys
- Adds an `UpdateSidebarAction`instead of the local storage for sharing
the state.

cathteng committed:
feat(aci): add `frequency` as optional key in workflow config schema (#82782)

scttcper committed:
test(ui): Disallow translations in tests (#82841)

ameliahsu committed:
feat(workflow-engine): add `FirstSeenEventConditionHandler` (#82822)

add `FirstSeenEventConditionHandler`

note that we need to add `Workflow` to `WorkflowJob` in order to get the
environment id

ryan953 committed:
ref(eslint): Rename and re-order config sections (#82845)

Running `npx eslint --inspect-config .` gives an easier to read list:

<img width="1218" alt="SCR-20250102-ntby"
src="https://github.com/user-attachments/assets/b113c2d2-abe4-49bd-8cb4-5e6805b38101"
/>

getsentry-bot committed:
Revert "ref(issue-views): Fold navigate behavior into issueViews context (#82540)"

This reverts commit 518b63d4f7c644d1c15b4e90c466dae127dffea6.

Co-authored-by: malwilley <10888943+malwilley@users.noreply.github.com>

evanpurkhiser committed:
Revert "fix(router): Add missing normalizeUrl to rr6 useNavigate (#82163)" (#82848)

This reverts commit 65a485715c0388fbe843c7d86a29e93b80ad2608.

ryan953 committed:
test: Enable some easy jest eslint rules and manually fix them (#82849)

asottile-sentry committed:
ref: fix parts of silo.test_client which were not running (#82789)

<!-- Describe your PR here. -->

JonasBa committed:
ref(tsc) fix no implicit any (#82825)

Fixes a couple instances where we were relying on implicit any

bcoe committed:
feat(insights): allow tooltip to be passed to transaction column headers (#82588)

nikkikapadia committed:
fix(widget-builder): Slideout content cut off on medium screen (#82823)

Noticed that the Add Widget/Update Widget button was being cut off on
medium sized screens. Changing the height to `fill-available` solved
this. It also corrects the positioning of the widget preview (apparently
it wasn't completely in the centre before 😒).

| Before | After |
|--------|--------|
| <img width="934" alt="image"
src="https://github.com/user-attachments/assets/40e3224f-4eba-4951-9a61-07dfee9ab728"
/> | <img width="934" alt="image"
src="https://github.com/user-attachments/assets/c249bbdd-4ecb-4815-9712-58d29ff8adb9"
/> |

narsaynorath committed:
fix(frames-delay): Update meta resolver frames delay units (#82209)

Fixes parsing the units for span metrics. This ensures that aggregates
over the span metrics will have the right units.

Fixes https://github.com/getsentry/sentry/issues/81560

---------

Co-authored-by: Markus Hintersteiner <markus.hintersteiner@sentry.io>

edwardgou-sentry committed:
feat(insights): handle missing vitals in tooltip weights (#82809)

Updates webvital score breakdown timeseries tooltip to display proper
weights when webvitals are missing

gggritso committed:
ref(dashboards): Remove Widget Viewer minimap feature (#82824)

This is effectively a revert of
https://github.com/getsentry/sentry/pull/34079/. We never shipped that
feature, and don't have plans to do so.

This PR reverts the major changes from that part:

1. Passing and handling `showSlider` around the Dashboard widgets and
the Widget Viewer modal
2. The `chartZoomOptions` prop that's only used in the Widget Viewer is
gone from both the component and hook version of chart zoom
3. The widget viewer modal state handling is reverted back to what it
was before
4. The "augmented" zoom handler is also gone

I left the `noPadding` prop alone. It's harmless, and partially in use.

natemoo-re committed:
fix(alerts): force re-render for stale RuleForm (#82842)

Closes #82335.

This PR "fixes" a state bug that occurs when an Alert is edited, saved,
and then edited again. It's an admittedly gross hack, but we're about to
completely refactor the `RuleForm` component as part of the Alerts
Create Issues (ACI) project!

MichaelSun48 committed:
ref(issue-views): Restructure issue views components  (#82593)

This PR reorganizes some of the components and logic in IssueViews
components. **No visual or functional changes should are intended by
these changes.** If you see any in the preview, please lmk.

The biggest changes include: 
* **Delete `draggableTabBar.tsx`** component, redistribute it's parts to
other components (detailed in the diff)
* **Create a new `IssueViewTab` component that contains the tab contents
(title, ellipsis menu, etc.).**
* Rename `draggableTabMenuButton.tsx` to `issueViewEllipsisMenu.tsx`
* Rename `draggableTabs` to `views` in `customViewsHeader.tsx`

Some reasoning: 
* `DraggableTabBar` sounds very redundant to and less clear than
`CustomViewsHeader` and `DraggableTabList`. To me, CustomViewsHeader
(soon to be renamed IssueViewsHeader) should be a specific
implementation of `DraggableTabList`, which sounds like it should be
reusable.
* I'd like to have parity between product names and codebase names -
hence why I'm changing "custom views" to "issue views" and "tabs" to
views.

Zylphrex committed:
fix: Missing optional chaining (#82860)

if `customStatsPeriod` is null/undefined, this will error.

Maybe fix JAVASCRIPT-2X8V

scttcper committed:
fix(issues): Move replay CTA into link, fix back button (#82850)

evanpurkhiser committed:
ref(insights): Simplify means of feature flag disabling tabs (#82477)

This is a follow up to @DominikB2014's comment here
https://github.com/getsentry/sentry/pull/82365#discussion_r1892587801

nikkikapadia committed:
feat(widget-builder): Validate widget before saving (#82807)

We're reusing the `validateWidget` function that is being used in the
current widget builder. I've added that validation check to the save
button so it will validate before carrying out the saving and closing
process. When the widget is not valid it displays the message as such:

<img width="1292" alt="image"
src="https://github.com/user-attachments/assets/78698bd0-186b-4454-a870-f5f3479ad5db"
/>

In future PRs I will implement field highlighting for the fields that
are causing validation errors.

Related to https://github.com/getsentry/sentry/issues/81729

ryan953 committed:
ref(eslint): Autofix some jest styleistic lint problems (#82851)

`yarn fix:eslint` is all.

asottile-sentry committed:
ref: another pass at deleting more dead test code (#82847)

<!-- Describe your PR here. -->

gggritso committed:
chore(dashboards): Remove widget viewer minimap feature declaration (#82863)

UI removed in https://github.com/getsentry/sentry/pull/82824

evanpurkhiser committed:
ref(js): Allow organizations actionCreators to use navigate (#82864)

michellewzhang committed:
ref(replay): add collapsible row to mobile tags (#82740)

- display the mobile redact classes in a prettier format by using
`CollapsibleValue`.
- kept the tooltip on overflow since we can't scroll the value

before:

<img width="635" alt="SCR-20241230-qhjv"
src="https://github.com/user-attachments/assets/fe935c02-21cb-4707-a139-1c4c4c503bce"
/>

after:


https://github.com/user-attachments/assets/fe97471b-ef3a-4417-847c-4d61f0aaa0e0

scttcper committed:
ref(performance): Remove unused event interfaces (#82866)

scttcper committed:
ref(issues): Remove unused stacktrace components (#82867)

evanpurkhiser committed:
ref(crons): Consistently import using relative imports (#82868)

Small clean up to make it easier to tease this apart to move things
around.

Part of GH-82428

ameliahsu committed:
feat(workflow-engine): add `NewHighPriorityIssueConditionHandler` (#82846)

add `NewHighPriorityIssueConditionHandler`

Commit:
chore(devservices): Cleanup unused docker compose testing files (#82871)

The scheduled devservices backend/acceptance jobs no longer use these
files

scttcper committed:
ref(issues): Clean up unused component exports (#82870)

JonasBa committed:
ref(tsc) enable nouncheckedindexedaccess (#82857)

Enable noUncheckedIndexedAccess

Fixes https://github.com/getsentry/frontend-tsc/issues/26

malwilley committed:
ref(issue-stream): Remove backend checks for issue-stream-performance flag (#82785)

JonasBa committed:
ref(tsc) fix noUncheckedIndexedAccess (#82876)

Merge race - this is now failing on master

scttcper committed:
ref(issues): Remove HoC from issue actions (#82877)

nikkikapadia committed:
fix(dashboards): Left alignment for edit access in table view (#82858)

Before rolling out I wanted to fix this small alignment issue that
Vasudha brought up. We wanted the access column to have the same
alignment as the rest of the columns. Before and after below :)

| Before | After |
|--------|--------|
| <img width="584" alt="image"
src="https://github.com/user-attachments/assets/dcee10d3-4f3c-4548-8b9d-4aa379ef0640"
/> | <img width="601" alt="image"
src="https://github.com/user-attachments/assets/10def1d4-d053-46c6-97b9-939cc27dfcb2"
/> |

(access column is properly left aligned)

scttcper committed:
ref(ui): Convert team members to FC (#82743)

evanpurkhiser committed:
fix(ui): Correct title consistencies in a number of views (#82879)

iamrajjoshi committed:
🔧 chore: remove redundant slack logging & metrics (#82744)

evanpurkhiser committed:
fix(ui): Title consistency in crons (#82882)

narsaynorath committed:
feat(widget-builder): Switching display type to big number (#82881)

Handles switching the display type to big number. To accomplish this, I
had to remove some fields from the UI because they aren't compatible
i.e. this commit does the following:

1. Only allow the user to select aggregates for the visualize step
2. Avoid displaying the alias field for big number widgets in the
visualize step
3. Update the hook to:
  - clear sort, yaxes, and legend aliases (there can only be one filter)
- set the fields to only the aggregates or yaxes (depending if you're
coming from table or chart)
- persist only the first filter query (big numbers don't support
multiple series)

Contributes to #81511

cathteng committed:
feat(aci): error detector (#82533)

narsaynorath committed:
fix(widget-builder): Field duplicating when toggling datasets (#82883)

If you set the big number display type, then switch to another dataset
and back, then y-axis gets populated because of the `else` statement
checking for `DisplayType.TABLE`. Then when you change the type to
`DisplayType.TABLE`, you'll see an extra `count()` being added.

I found the same behaviour occurs when switching datasets with a
line chart displayed, this is because we were setting `fields` and then
setting the y-axis, but when it's a timeseries chart we should be
clearing the `fields` state.

ryan953 committed:
fix(hydration issue): Fix button spacing on hydration diff modal (#82885)

**Before:**
<img width="379" alt="SCR-20250103-llas"
src="https://github.com/user-attachments/assets/92822fce-8c82-403b-a585-c1fefa8234a9"
/>

**After:**
(simulated two buttons in dev)

<img width="385" alt="SCR-20250103-lkzx"
src="https://github.com/user-attachments/assets/840a0c79-1865-46fe-aede-d1d9fde02084"
/>

ryan953 committed:
ref(eslint): Import many recommended rulesets, but disable whatever we are violating (#82872)

I removed a bunch of rules which are off by default
Then brought in recommended rulesets, and removed rules which are
included in there. But some of the recommended rules are problems right
now, so i explicitly disabled those and added a TODO comment

The rules are organized like:
```
rules: {
  // rules that are not in recommended set
  ...plugin.configs.recommended.rules,
  // disable something that's recommended, ideally temporary until we can fix it
}
```

I also got rid of all the inline comments that point to individual rule
docs, there were a lot and it made the file longer imo. Instead I added
links to the full rule list for each plugin. Linked to either the readme
or code, depending on what's available. And for any recommended ruleset
i linked to that list directly as well. So it's easy to open up links
for a specific plugin and see what's available, and also what's enabled.

🐎 Some new rules are enabled with this change, and we're not violating
any of them! This should affect runtime right? I've noticed that before
we were in the ~55s area for `yarn lint:js`, and with this it's ~60s.
More rules to enable in future. I think that the increase so far isn't
significant.

gggritso committed:
ref(dashboards): Eliminate duplication of `LineChartWidget` and `AreaChartWidget` code (#82835)

The only meaningful differences were:

- the timeseries are constructed with `AreaSeries` vs. `LineSeries`
- completeness delay was not supported

I added support for completeness delay, passed the timeseries
constructors as a prop, and moved all the shared code into a
`TimeSeriesVisualization` component. No fuss, no muss.

asottile-sentry committed:
ref: remove ui for monitor checkin attachments (#82890)

datadog confirms this is unused, cleaning up the frontend first!

<!-- Describe your PR here. -->

iamrajjoshi committed:
:wrench: chore: remove msteams tenant ff handler (#82886)

c298lee committed:
fix(replay): Fix video flickering on Safari (#82769)

To prevent flickering in Safari, we set the z-index of the clip
depending on its index. We hide all videos except the next video and the
video before that. This "fixes" the gap in the video by showing the
previous video when the gap is happening.

Example replay:
https://brustolin.sentry.io/replays/c817d3d7daac4440a2788fd0cbf521b0

Before:
Playthough with gaps


https://github.com/user-attachments/assets/49e19a85-8e38-4f03-900b-e57202db88d5

Clicking to gap
<img width="1505" alt="image"
src="https://github.com/user-attachments/assets/fd130c1d-6de1-4651-abf5-ef8d4d56d536"
/>


After:
Playthrough is smooth


https://github.com/user-attachments/assets/83db4493-fcc9-4fbc-87c3-cdccad3df823

No empty video at same timestamp
<img width="1505" alt="image"
src="https://github.com/user-attachments/assets/a1f3c6de-8607-4a38-a118-7d54a9ab72cd"
/>

Fixes https://github.com/getsentry/sentry/issues/82771

narsaynorath committed:
fix(widget-builder): Reset sort when changing display type to table (#82889)

Noticed that tables weren't resetting the sort to a valid value (i.e. we
have to sort by a field that's selected), so when going from a
timeseries chart to a table, we should reset the sort to the first
column if the current sort is not in the table, otherwise it can stay
the same.

Note: I found another bug that's similar that I will fix in a following
PR. The sort doesn't reset when you remove a field that is being sorted
on in a table

ameliahsu committed:
feat(workflow-engine): add `LevelConditionHandler` (#82874)

add `LevelConditionHandler`

wedamija committed:
fix(uptime): Fix repair task to successfully repair update tasks (#82896)

Missed this when adding updates

<!-- Describe your PR here. -->

ryan953 committed:
ref(replay): Update hydration Before/After labels to have timestamps in their tooltip (#82895)

asottile-sentry committed:
ref: remove unused id-or-slug option override (#82902)

<!-- Describe your PR here. -->

roggenkemper committed:
chore(issue-details): Adjust min width for tags (#82905)

this pr updates the min width for tags to allow for a wider range of
widths to better differentiate values in the breakdown

before: 
<img width="105" alt="Screenshot 2025-01-03 at 2 34 01 PM"
src="https://github.com/user-attachments/assets/5fd19144-4bd3-4d1c-b66d-ed62f8a6e4e9"
/>

after:
<img width="173" alt="Screenshot 2025-01-03 at 2 34 07 PM"
src="https://github.com/user-attachments/assets/39b20381-6f5a-410c-84e9-6f10566e9c2a"
/>

ryan953 committed:
fix(hydration issue): Fix tooltip position for test copy button (#82912)

**Before**
both tooltips were active at the same time:

![SCR-20250103-nqhc](https://github.com/user-attachments/assets/4395eadf-e0fa-455f-a66f-3e9c7335840a)

**After**
one at a time
<img width="137" alt="SCR-20250103-nqiu"
src="https://github.com/user-attachments/assets/b147f3f3-eaa7-42d7-a990-6c959ea3066f"
/>

ryan953 committed:
fix(hydration issue): Only look at IncrementalSource.Mutation types for hydration before/after (#82908)

There are many IncrementalSource possibilities (definition copied
below), but we really only want to look at mutations. Mouse moves,
input, and possibly even StyleSheetRule shouldn't affect hydrating the
DOM that react cares about.

The full list of this enum is:
```
export declare enum IncrementalSource {
    Mutation = 0,
    MouseMove = 1,
    MouseInteraction = 2,
    Scroll = 3,
    ViewportResize = 4,
    Input = 5,
    TouchMove = 6,
    MediaInteraction = 7,
    StyleSheetRule = 8,
    CanvasMutation = 9,
    Font = 10,
    Log = 11,
    Drag = 12,
    StyleDeclaration = 13,
    Selection = 14,
    AdoptedStyleSheet = 15,
    CustomElement = 16
}
```

The parent type that we are refining down from is:
```
export type incrementalSnapshotEvent = {
    type: EventType.IncrementalSnapshot;
    data: incrementalData;
};
```

bcoe committed:
feat(insights): add tooltips user misery and apdex (#82880)

michellewzhang committed:
ref(replay): rm experimental from onboarding (#82655)

closes https://github.com/getsentry/sentry/issues/82255

---------

Co-authored-by: Bruno Garcia <bruno@brunogarcia.com>

asottile-sentry committed:
ref: fix typing for serializers.models.dashboard (#82859)

<!-- Describe your PR here. -->

Zylphrex committed:
fix(profiling): View profile links from trace drawer (#82893)

The current view only links for continous profiles. Make it work for
transaction profiles as well. The new view links don't work at all.

Zylphrex committed:
fix(profiling): Link profiler id in context to flamegraph (#82901)

Continuous profiles use the profiler id in the profile context and we
should try to link that to the flamegraph wherever possible as well.

cmanallen committed:
feat(replays): Add billing skip option for beta organizations (#82924)

Precursor to:
https://github.com/getsentry/sentry-options-automator/pull/2941

gggritso committed:
feat(insights): Use `GlobalDrawer` instead of detail panels (#82534)

Closes https://github.com/getsentry/sentry/issues/81273

Replaces all Insights side panels with new `GlobalDrawer`
implementations that match the drawers in Issues. The visual changes are
minimal

1. Slightly different animations
2. A "Close" button on the top left of the drawer
3. Scroll bars on the left of the drawer
4. Tighter padding
5. Panel doesn't re-animate as often on clicking another transaction

There are small feature changes!

- no more docking bottom or right. Docking is to the right only
- panels now have closing animations

## Code Changes

`GlobalDrawer` is invoked with a hook, rather than rendered inline. So,
I had to make some changes.

**Before:**

- rendering panels mostly unconditionally like `<CacheSamplesPanel />`
- each sample panel checks the URL to see if it's supposed to be open
- if open, renders itself, otherwise stays invisible
- on close, updates the URL, which re-renders and hides

This is problematic for a bunch of reasons:

1. When components can decide to not render themselves, the code flow is
confusing. Parents should generally decide when to render children
2. When a component is rendered but hidden, it runs its side effects.
This means that in our case, network requests were sometimes going out,
even though the panel isn't visible

**After:**

- the panel is rendered using a `useEffect`, the effect detects a
necessary URL parameter or state, and open the panel if necessary. All
conditions that the panel used to check for itself are pulled up into
the parent
- the panel is rendered straight, it doesn't have any conditionals to
check if it's supposed to be open
- on close, the panel hides itself, and tells the parent to update the
URL or state as necessary

This is tidier, avoids component side-effects, and preserves closing
animations! Plus, this means I can re-use a header component, make the
analytics tracking consistent, etc.

---

This only leaves a small handful of users of `DetailPanel`, mostly in
the trace view.

cmanallen committed:
feat(replays): Add option to control who is and isn't billed for mobile replays (#82680)

Do not merge until the option is defined in the options repo.

Closes: https://github.com/getsentry/team-replay/issues/523

---------

Co-authored-by: Bruno Garcia <bruno@brunogarcia.com>

Zylphrex committed:
fix(discover): Format ios device names (#82927)

This ensures discover renders device names as their human readable
equivalent.

cmanallen committed:
feat(replays): Emit metric on mobile-replay billing skip (#82928)

Co-authored-by: Bruno Garcia <bruno@brunogarcia.com>

edwardgou-sentry committed:
feat(insights): Add flag for handling standalone lcp and cls spans  (#82929)

Zylphrex committed:
chore: Delete some unused code in events filter (#82760)

This is referring to the old json style snuba queries that we don't use
anymore.

mrduncan committed:
fix(issues): Remove duplicate span in update_existing_attachments (#82911)

This is already auto-instrumented in run_post_process_job with the exact
same span op.

See also: #82910

mrduncan committed:
fix(issues): Remove duplicate span in sdk_crash_monitoring (#82910)

This op is already being auto-instrumented in run_post_process_job, the
duplicate span here with the same name is not helpful when looking at
aggregates.

This renames the op and adds another around config building - in the
future I suspect we can get a very easy win by not rebuilding config for
every single event - this will give us some data to back up that theory.

Aggregate showing the 2x duplicate (`update_existing_attachments` is
fixed in a separate PR):
https://sentry.sentry.io/traces/?groupBy=span.op&mode=aggregate&project=1&query=span.op%3Atasks.post_process_group.%2A&statsPeriod=24h&visualize=%7B%22chartType%22%3A1%2C%22yAxes%22%3A%5B%22count%28span.duration%29%22%5D%7D

See also: #82911

malwilley committed:
ref(issue-list): Create copy of overview.tsx as a functional component (#82784)

Adds a copy of overview.tsx (and tests) that is a functional component.
I mostly copied it 1 for 1, so the resulting component is not much
better than the original. We need a starting place first to start making
more improvements.

This new component is only rendered if the
`issue-stream-functional-refactor` flag is enabled. We will test this
internally before rolling out more widely and deleting the original to
reduce risk.

The most significant changes were around data fetching and caching,
since converting things to useEffects resulted in some bugs and test
failures that the original lifecycle methods managed to avoid somehow.
For example, we are now using the `requestParams` object to determine if
we should refetch instead of checking against previous props for
everything individually.

I have also removed `issue-stream-performance` checks in this version of
the component since that complicated things significantly.

ameliahsu committed:
feat(workflow-engine): add `TaggedEventConditionHandler` (#82900)

add `TaggedEventConditionHandler`

iamrajjoshi committed:
:bug: fix(devserver): Update ngrok configuration for monolith mode (#82906)

iamrajjoshi committed:
:wrench: chore(api): cleanup ownership allowlist (#82915)

getsentry-bot committed:
Revert "ref(issue-list): Create copy of overview.tsx as a functional component (#82784)"

This reverts commit 7c678ff11fe4803fa0b46ae381f4e5ee722646d5.

Co-authored-by: malwilley <10888943+malwilley@users.noreply.github.com>

ryan953 committed:
ref(hydration error): Refactor the hydration modal to have a shared context for before/after timestamps (#82888)

This shared context will help to centralize the before/after timestamps
and stops us from having to do prop-drilling to get these values down
into all the places they're used.

The Provider is added to the section inside the Issue Details page,
where `<ReplaySliderDiff>` is rendered unconditionally.
The timestamps in use there are then sent into the modal, where another
Provider is created to wrap all the tabs inside the hydration modal.

There are unused setters in the Provider right now. If a setter inside
the modal is used, it won't affect the diff on Issue Details page (which
is how the modal can be triggered). But all tabs within the modal will
share the same before/after timestamps.

scttcper committed:
feat(issues): Preserve group when switching environment (#82907)

michellewzhang committed:
ref(replay): rm beta badges and add grace period banner for mobile replay (#82467)

- closes https://github.com/getsentry/sentry/issues/77646
- closes https://github.com/getsentry/sentry/issues/82651

to be merged closer to GA date (jan 6)

added the grace period banner to this PR:

<img width="1269" alt="SCR-20241230-ksjo"
src="https://github.com/user-attachments/assets/e4877671-a4fe-4aa4-9deb-00238e86cb75"
/>

---------

Co-authored-by: Bruno Garcia <bruno@brunogarcia.com>

Zylphrex committed:
chore(explore): Add extrapolation meta as valid referrer (#82932)

JonasBa committed:
ref(storybook) use react query for module loading (#82931)

Use react query for story loading

malwilley committed:
ref(issue-list): Create copy of overview.tsx as a functional component (#82935)

Same as https://github.com/getsentry/sentry/pull/82784 which got
reverted, with the addition of a type failure fix.

Adds a copy of overview.tsx (and tests) that is a functional component. I mostly copied it 1 for 1, so the resulting component is not much better than the original. We need a starting place first to start making more improvements.

This new component is only rendered if the issue-stream-functional-refactor flag is enabled. We will test this internally before rolling out more widely and deleting the original to reduce risk.

The most significant changes were around data fetching and caching, since converting things to useEffects resulted in some bugs and test failures that the original lifecycle methods managed to avoid somehow. For example, we are now using the requestParams object to determine if we should refetch instead of checking against previous props for everything individually.

I have also removed issue-stream-performance checks in this version of the component since that complicated things significantly.

michellewzhang committed:
fix(replays): fix placement of mobile banner (#82936)

## AFTER:

under ff (is a beta org) and proj is mobile replay platform:

<img width="1181" alt="SCR-20250106-jexg"
src="https://github.com/user-attachments/assets/aadea33f-99e0-4383-9cbe-1d050aaccbe4"
/>

under ff but proj is not a replay mobile platform:

<img width="1186" alt="SCR-20250106-jewc"
src="https://github.com/user-attachments/assets/30285401-194a-4bef-bbef-41e7f45df732"
/>

## BEFORE: 

under ff (is a beta org) and proj is mobile replay platform:


<img width="1184" alt="SCR-20250106-jfpb"
src="https://github.com/user-attachments/assets/8df07232-8a2a-49f1-9ea4-f61139916223"
/>

under ff but proj is not a replay mobile platform:


<img width="1178" alt="SCR-20250106-jfqf"
src="https://github.com/user-attachments/assets/75d5693e-8f24-40d1-a110-4ccd044f1a58"
/>

nikkikapadia committed:
feat(widget-builder): Add error field in WidgetBuilderState (#82887)

Add an error field to the widget builder state. It is just a regular
state, not a queryParamState so it is not stored in the URL. This will
be used to keep track of the errors we get on validation so we can
highlight the correct fields to prompt users to change the input. This
PR is the baseline to set up that functionality.

Contributes to https://github.com/getsentry/sentry/issues/81729

untitaker committed:
chore(snuba): Fix testcase to account for stricter Rust errors processor (#82934)

The Rust errors processor in Snuba has been in prod for a long time now,
but has never been used in tests. Update testcase so that we can get rid
of the Python one.

scttcper committed:
chore(insights): Remove unused useErrorRateQuery (#82938)

scttcper committed:
chore(webvitals): Remove unused Recommendations (#82939)

narsaynorath committed:
fix(widget-builder): Update table sort when field removed or modified (#82898)

When a field is removed or modified, the sort field should update
accordingly. This relies on the fact that we're only going to update one
step at a time (i.e. we won't update two fields through the hook in one
action) and determines if:

a) a field was removed
b) a field was modified

When a field was removed, we simply reset to the first option.

When a field is modified, we look for which field that was and then
update the sort to reflect that field.

I had to fix a bug in the visualize component that was updating the
state directly (i.e. directly modifying the memory) instead of updating
it through the dispatch method. I fixed this by doing a deep clone and
updating the field then calling the dispatch function.

scttcper committed:
chore(performance): Remove unused span count histogram query (#82940)

shruthilayaj committed:
fix(discover): Fix tag facet breakdown for errors (#82930)

Fixes https://github.com/getsentry/sentry/issues/82149

Make events-facets endpoint dataset aware. Fixes the
issue where tag breakdown numbers were funky for
errors.

michellewzhang committed:
fix(replay): banner logic was in wrong spot (#82942)

getsentry-bot committed:
Revert "feat(insights): Use `GlobalDrawer` instead of detail panels (#82534)"

This reverts commit 81e09187e94f47885dd98034f80f85f2b661da11.

Co-authored-by: gggritso <989898+gggritso@users.noreply.github.com>

narsaynorath committed:
fix(widget-builder): Bypass sort reset for issue widgets (#82944)

Bypasses the sorting reset for issue widgets. This is because they use a
different API and they don't need to enforce that the field needs to be
in the table.

evanpurkhiser committed:
ref(js): Remove most deprecatedAsyncView usages (#82913)

Co-authored-by: Michelle Zhang <56095982+michellewzhang@users.noreply.github.com>

evanpurkhiser committed:
ref(ui): Use AsyncComponent over AsyncView when appropriate (#82899)

JonasBa committed:
ref(colors) make color pallete readonly (#82922)

Use readonly so we dont end up mutating the colors by mistake
(dangerously close in two places). Using a const arr means we can also
drop the non null assertions and guard from out of bounds access.

The getColorPallete should also return a T | undefined as there is
nothing enforcing access inside the array indices

Christinarlong committed:
chore(sentry apps): Deprecate send_alert_event (#82947)

michellewzhang committed:
fix(replay): flag option needs api expose (#82946)

edwardgou-sentry committed:
feat(insights): update total opportunity score function to handle missing vitals (#82923)

Updates total opportunity score function to upscale if there are any
missing webvitals score. Consolidates the total weight logic in
performance and opportunity scores into a single
`_resolve_total_weights_function` function.

Abdkhan14 committed:
feat(explore): Adding perf unit quota flag to metadata analytics event. (#82950)

Co-authored-by: Abdullah Khan <abdullahkhan@PG9Y57YDXQ.local>

ceorourke committed:
chore(alerts): Remove fe code for activated alerts (#81218)

Remove the front end for activated alerts as it's being deprecated. I
can't merge this until we clean up a couple live alerts but I wanted to
get the PRs ready. The gigantic backend counterpart is here
https://github.com/getsentry/sentry/pull/81095

**Before**
<img width="1096" alt="Screenshot 2024-11-25 at 1 25 54 PM"
src="https://github.com/user-attachments/assets/8d3fdbc2-59d9-4c38-9032-cf89f56e9153">


**After**
<img width="1078" alt="Screenshot 2024-11-25 at 1 27 26 PM"
src="https://github.com/user-attachments/assets/6d7d5191-285d-465a-9eda-a8441958e8bd">

narsaynorath committed:
fix(widget-builder): Use first widget query for shared fields (#82953)

Widget query properties such as fields, columns, aggregates are shared
so when we `flatMap` we duplicate the values. We should just choose the
first one and use that to avoid duplication.

Closes #82548

nikkikapadia committed:
feat(widget-builder): Implement error message for title field (#82948)

Added an error message to the name field to handle validation errors on
submit. I've just repurposed the type of field that we use on the
current widget builder as it has the error message functionality built
in. When the user starts typing in that field it will get rid of the
error.

Looks like this :
<img width="724" alt="image"
src="https://github.com/user-attachments/assets/b8ac2ec2-aed4-4495-ae00-6d10fc7dea34"
/>

Contributes to https://github.com/getsentry/sentry/issues/81729

ryan953 committed:
feat(hydration error): Iterate on formatting of mutation diff data (#82954)

| Before | After |
| --- | --- |
| <img width="679" alt="SCR-20250106-jehx"
src="https://github.com/user-attachments/assets/2e61f5a3-f63e-414d-bccc-a4771ec104e9"
/> | <img width="682" alt="SCR-20250106-ljwv"
src="https://github.com/user-attachments/assets/8ce7f3d2-63a9-46f6-8977-7a6ff1ff0a41"
/> |

notice that the key is the formatted offset within the replay.
The raw timestamp is moved into the payload, next to the numeric offset.

wmak committed:
feat(rpc): Run queries in parallel (#81837)

- This adds a way to run table and timeseries queries in parallel, these
are the two we're most likely to need to run in parallel, but we can
extend this to the other rpc queries later as needed
- TODO: not a blocker for this PR i don't think, but I need to look into
whether we still need all this SDK scope stuff, i'm just copy pasting
from the existing snql parallel code

ameliahsu committed:
feat(workflow-engine): add `AgeComparisonConditionHandler` (#82933)

scttcper committed:
feat(settings): Convert team notifications to FC (#82894)

evanpurkhiser committed:
ref(rr6): Rename RouteContext to TestRouteContext for clarity (#82951)

We don't want people accidentally misusing this, make it clear this is
JUST for testing at the moment

ceorourke committed:
chore(alerts): Remove org alert rule activations endpoint (#82955)

redo of https://github.com/getsentry/sentry/pull/81281 which somehow got
borked from being stale

narsaynorath committed:
fix(widget-builder): Deep clone new field so updates don't alias (#82957)

When adding new fields, we were just adding the default query each time
which would always point to the same memory address. We need to clone it
so each additional field is treated individually and modifying the
fields array doesn't trickle changes to the other added fields because
they reference the same object.

Closes #82945

ceorourke committed:
chore(alerts): Remove activated alerts from serializers (#81282)

Another chunk of https://github.com/getsentry/sentry/pull/81095/files
that removes activated alerts from serializers.

This can't be merged until
https://github.com/getsentry/sentry/pull/81218 is merged.

nikkikapadia committed:
feat(widget-builder): Add error functionality to type selector (#82958)

Shows an error beside the widget type selector if there is an error. The
way we have the widget builder set up right now there should be no error
that arises however it's good to have the error indicator/message there
anyways.

This is what it would look like if an error arises: 
<img width="594" alt="Screenshot 2025-01-06 at 4 07 43 PM"
src="https://github.com/user-attachments/assets/22a66fad-4955-4a96-8418-71785903eb84"
/>

Contributes to https://github.com/getsentry/sentry/issues/81729

aliu39 committed:
feat(feedback): add a trace.id tag during ingestion (#81875)

This is a useful tag to see without clicking through to the full trace
via trace section. There are also some cases we hide the trace section
(crash report dup), but this tag will show up no matter what.

MichaelSun48 committed:
ref(issue-views): Minor renaming and clean up changes (#82904)

Makes a couple of small, non-visual changes to issue views components: 

1. Renames the directory containing issueviews components from
`groupSearchViews` -> `issueViews`
2. Replaces `organization` prop with `useOrganization` in a couple
places

scttcper committed:
fix(settings): Correct settings/members/requests redirect (#82962)

scttcper committed:
fix(alerts): Remove query from metric alert related issues (#82965)

snigdhas committed:
feat(metric-issues): Disable manual resolution for metric issues (#82937)

Disables manual resolution for metric issues and updates relevant copy
to match the
[mocks](https://www.figma.com/design/p2DOzd1RwAHQbq0v6KaYmF/Q2-Issue-Details?node-id=4758-53209&t=uxzR2rduAuncrceP-0)

<img width="1286" alt="image"
src="https://github.com/user-attachments/assets/70a31f94-d21e-4299-8273-2e643af2f1b3"
/>


<img width="1288" alt="image"
src="https://github.com/user-attachments/assets/493b7d9d-f5d4-4506-849b-002be2829f03"
/>

cathteng committed:
feat(aci): issue alert dual write helpers (#82818)

ameliahsu committed:
ref(crons): new `mergeBuckets` timeline function without env (#82569)

new `mergeBuckets` function that merges new `monitorBucketWithStats`
type

since `checkinTimeline` doesn't need to know anything about environment,
we want to replace `MonitorBucketEnvMapping` fields with `StatsBucket`

ceorourke committed:
chore(alerts): Remove activated alerts from notifications (#81283)

Another chunk of https://github.com/getsentry/sentry/pull/81095 that
removes the notification code for activated alerts.

This can't be merged until
https://github.com/getsentry/sentry/pull/81218 is merged.

ameliahsu committed:
fix(crons): remove export from test (#82975)

fix linting error by removing export from test file

https://github.com/getsentry/sentry/actions/runs/12641932158
